src/categorical_algebra/CSets.jl

```diff
@@ -352,4 +356,29 @@ function (::Type{T})(X::ACSet, FOb::AbstractDict,
   migrate!(Y, X, FOb, FHom)
 end
 
+""" Serialize an ACSet object to a JSON string
+"""
+serialize(x::ACSet) = JSON.json(x.tables)
+
+""" Deserialize a dictionary from a parsed JSON string to an object of the given ACSet type
+"""
+function deserialize(input::Dict, type)
```

Comment:
This is a bike-shed comment, but IMO the type should be the first argument, so that the signature is:

```julia
parse_json_acset(::Type{T}, input::AbstractDict) where T <: AbstractACSet
```

Reply:
Moved the type argument position and cleaned up type definitions